crates/storage/provider/src/transaction.rs

```diff
@@ -510,6 +510,49 @@ where
         Ok(())
     }
 
+    /// Append blocks and insert its post state.
+    /// This will insert block data to all related tables and will update pipeline progress.
+    pub fn append_blocks_with_post_state(
```

Comment:
I would like to get a test on this since I tried to do something similar but it failed during sync at some points, with the merkle parts complaining. Just a simple one making sure that it actually completes the write for a range of blocks or something

Reply:
It is kinda tricky to do as changesets need to match previous value of the parent block, as if it does not match get/take would not work correctly and you would get a different Merkle root.

Added test that inserts two blocks, gets one, gets two, takes two.